server/src/controllers/saved-package.controller.ts

```diff
@@ -0,0 +1,76 @@
+import { Request, Response } from 'express';
+import { SavedPackageRepository } from '../repositories/saved-packages.repository';
+import mongoose from 'mongoose';
+
+export const savedPackageController = {
+    getUsersSavedPackages: async (req: Request, res: Response) => {
+        try {
+            const { packageId } = req.query;
+
+            const matchStage: Record<string, any> = {
+                userId: new mongoose.Types.ObjectId(req.userId),
+            };
+
+            if (packageId) {
+                matchStage.packageId = new mongoose.Types.ObjectId(packageId as string);
+            }
+
+            const pipeline: mongoose.PipelineStage[] = [
+                { $match: matchStage },
```

Comment:
```suggestion
          const matchStage: SavedPackage = {
                userId: new mongoose.Types.ObjectId(req.userId),
                matchStage.packageId = new mongoose.Types.ObjectId(packageId as string);
            };
            
            // rest here and then
                { $match: matchStage },
```
make that in `saved-package.service.ts`